frontend/src/components/bookingView/BookingViewSlotSelection.vue

```diff
@@ -1,3 +1,51 @@
+<script setup lang="ts">
+import { inject } from 'vue';
+import { storeToRefs } from 'pinia';
+import { useI18n } from 'vue-i18n';
+
+import { useBookingViewStore } from '@/stores/booking-view-store';
+import { useScheduleStore } from '@/stores/schedule-store';
+import { dateFormatStrings } from '@/definitions';
+import { Slot } from '@/models';
+import { dayjsKey } from "@/keys";
+
+import PrimaryButton from '@/elements/PrimaryButton.vue';
+import CalendarQalendar from '@/components/CalendarQalendar.vue';
+
+const { t } = useI18n();
+const { activeSchedules } = storeToRefs(useScheduleStore());
+const { appointment, activeDate, selectedEvent } = storeToRefs(useBookingViewStore());
+const dj = inject(dayjsKey);
+
+const emit = defineEmits(['openModal']);
+
+// component properties
+interface Props {
+  showNavigation: boolean,
+};
+defineProps<Props>();
+
+/**
+ * Select a specific time slot
+ * @param day string
+ */
+const selectEvent = (day: String) => {
+  // set event selected
+  for (let i = 0; i < appointment.value.slots.length; i += 1) {
+    const slot: Slot = appointment.value.slots[i];
+    if (slot.start.format(dateFormatStrings.qalendar) === day) {
+      slot.selected = true;
+      const e = { ...appointment.value, ...slot };
+      delete e.slots;
+      selectedEvent.value = e;
```

Comment:
For now I typed that as `Appointment & Slot`, but we might want to evaluate if we even need an intersection type here.